main.nf

```diff
@@ -518,6 +519,9 @@ if(params.pseudo_aligner == 'salmon' && !params.salmon_index){
             publishDir path: { params.saveReference ? "${params.outdir}/reference_genome" : params.outdir },
                                saveAs: { params.saveReference ? it : null }, mode: 'copy'
 
+            when:
+            !params.transcript_fasta
```

Comment:
@olgabot This evaluation is already performed with the `if` statement above. Is that not working?